src/components-styled/chart-tile.tsx

```diff
@@ -0,0 +1,105 @@
+import css from '@styled-system/css';
+import { useState } from 'react';
+import { TimeframeOption } from '~/utils/timeframe';
+import { Box } from './base';
+import { ChartTileContainer } from './chart-tile-container';
+import { ChartTimeControls } from './chart-time-controls';
+import { MetadataProps } from './metadata';
+
+interface ChartTileProps {
+  children: React.ReactNode;
+  metadata: MetadataProps;
+  title: string;
+  description?: React.ReactNode;
+  showDataWarning?: boolean;
+}
+
+interface ChartTileWithTimeframeProps extends Omit<ChartTileProps, 'children'> {
+  children: (timeframe: TimeframeOption) => React.ReactNode;
+  timeframeOptions?: TimeframeOption[];
+  timeframeInitialValue?: TimeframeOption;
+}
+
+export function ChartTile({
+  title,
+  description,
+  metadata,
+  children,
+  showDataWarning,
+}: ChartTileProps) {
+  return (
+    <ChartTileContainer metadata={metadata} showDataWarning={showDataWarning}>
+      <ChartTileHeader title={title} description={description} />
+      {children}
+    </ChartTileContainer>
+  );
+}
+
+export function ChartTileWithTimeframe({
+  title,
+  description,
+  metadata,
+  timeframeOptions = ['all', '5weeks', 'week'],
+  timeframeInitialValue = '5weeks',
+  showDataWarning,
+  children,
+}: ChartTileWithTimeframeProps) {
+  const [timeframe, setTimeframe] = useState<TimeframeOption>(
+    timeframeInitialValue
+  );
+
+  return (
+    <ChartTileContainer metadata={metadata} showDataWarning={showDataWarning}>
+      <ChartTileHeader
+        title={title}
+        description={description}
+        timeframe={timeframe}
+        timeframeOptions={timeframeOptions}
+        onTimeframeChange={setTimeframe}
+      />
+      {children(timeframe)}
+    </ChartTileContainer>
+  );
+}
+
+function ChartTileHeader({
+  title,
+  description,
+  timeframe,
+  timeframeOptions,
+  onTimeframeChange,
+}: {
+  title: string;
+  description?: React.ReactNode;
+  timeframe?: TimeframeOption;
+  timeframeOptions?: TimeframeOption[];
+  onTimeframeChange?: (timeframe: TimeframeOption) => void;
+}) {
+  return (
+    <Box
+      mb={3}
+      display="flex"
+      flexDirection={['column', null, null, null, 'row']}
```

Comment:
why not use the named versions for the mediaqueries?
```suggestion
      flexDirection={{ xs: 'column', xl: 'row' }}
```